hibernate-core/src/main/java/org/hibernate/query/internal/QueryParameterBindingsImpl.java

```diff
@@ -166,45 +167,25 @@ public void visitBindings(@SuppressWarnings("rawtypes") BiConsumer action) {
 
 	@Override
 	public QueryKey.ParameterBindingsMemento generateQueryKeyMemento(SharedSessionContractImplementor persistenceContext) {
-		final int size = parameterBindingMap.size();
-		final List<Object> allBindValues = new ArrayList<>( size );
-		int hashCode = 0;
+		final MutableCacheKeyImpl mutableCacheKey = new MutableCacheKeyImpl();
```

Comment:
Pre-sizing the `ArrayList` might be nice

```suggestion
		final MutableCacheKeyImpl mutableCacheKey = new MutableCacheKeyImpl( parameterBindingMap.size() );
```